plugin/src/main/java/org/opensearch/ml/processor/MLInferenceIngestProcessor.java

```diff
@@ -0,0 +1,253 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.opensearch.ml.processor;
+
+import static org.opensearch.ml.common.utils.StringUtils.gson;
+import static org.opensearch.ml.processor.MLModelUtil.*;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.function.BiConsumer;
+
+import org.opensearch.action.ActionRequest;
+import org.opensearch.client.Client;
+import org.opensearch.core.action.ActionListener;
+import org.opensearch.core.common.Strings;
+import org.opensearch.ingest.AbstractProcessor;
+import org.opensearch.ingest.ConfigurationUtils;
+import org.opensearch.ingest.IngestDocument;
+import org.opensearch.ingest.Processor;
+import org.opensearch.ingest.ValueSource;
+import org.opensearch.ml.common.output.model.ModelTensorOutput;
+import org.opensearch.ml.common.transport.MLTaskResponse;
+import org.opensearch.ml.common.transport.prediction.MLPredictionTaskAction;
+import org.opensearch.script.ScriptService;
+import org.opensearch.script.TemplateScript;
+
+public class MLInferenceIngestProcessor extends AbstractProcessor implements ModelExecutor {
+
+    private final MLModelUtil mlModelUtil;
+    private final boolean ignoreMissing;
+    private final ScriptService scriptService;
+    private static Client client;
+    public static final String TYPE = "ml_inference";
+    public static final String DEFAULT_OUTPUT_FIELD_NAME = "inference_results";
+
+    protected MLInferenceIngestProcessor(
+        String model_id,
+        List<Map<String, String>> input_map,
+        List<Map<String, String>> output_map,
+        Map<String, String> model_config,
+        String tag,
+        String description,
+        boolean ignoreMissing,
+        ScriptService scriptService,
+        Client client
+    ) {
+        super(tag, description);
+        this.mlModelUtil = new MLModelUtil(model_id, input_map, output_map, model_config);
```

Comment:
In java, Util class usually are classes only have static methods and don't have state in it and we don't inject it to other classes. Instead we use Util class directly with static method call. Please either change the MLModelUtil to a class with static methods or rename this class.

Reply:
renamed to InferenceProcessorAttributes